pkg/compose/compose.go

```diff
@@ -63,6 +66,25 @@ func (s *composeService) MaxConcurrency(i int) {
 	s.maxConcurrency = i
 }
 
+func (s *composeService) DryRunMode(dryRun bool) error {
+	if dryRun {
+		cli, err := command.NewDockerCli()
+		if err != nil {
+			return err
+		}
+		err = cli.Initialize(flags.NewClientOptions(), command.WithInitializeClient(func(cli *command.DockerCli) (client.APIClient, error) {
+			dryRunClient := api.NewDryRunClient()
+			dryRunClient.WithAPIClient(s.apiClient())
+			return dryRunClient, nil
+		}))
+		if err != nil {
+			return err
+		}
+		s.dockerCli = cli
```

Comment:
could make it simpler by making API client an attribute in `composeService`, initialize with `dockerCli.Client()` and override here with `NewDryRunClient`.

Reply:
Ok after digging a little bit, the `dockerCli` attribute is used widely by the CLI APIs, so it seams safer to override the APIClient of the existing `command.Cli`